boilerplates/lucia-auth/files/server/lucia-auth-handlers.ts

```diff
@@ -0,0 +1,341 @@
+   */
+  await lucia.invalidateSession(session.id);
+
+  /**
+   * Delete session cookie
+   *
+   * @link {@see https://lucia-auth.com/basics/sessions#delete-session-cookie}
+   */
+  return new Response(JSON.stringify({ status: "success" }), {
+    status: 200,
+    headers: {
+      "set-cookie": lucia.createBlankSessionCookie().serialize(),
+    },
+  });
+}
+
+/**
+ * Github OAuth authorization handler
+ *
+ * @link {@see https://lucia-auth.com/guides/oauth/basics#creating-authorization-url}
+ */
+export async function luciaGithubLoginHandler<Context extends Record<string | number | symbol, unknown>>(
+  _request: Request,
+  _context?: Context,
+): Promise<Response> {
+  const state = generateState();
+  const url = await github.createAuthorizationURL(state);
+
+  return new Response(null, {
+    status: 302,
+    headers: {
+      Location: url.toString(),
+      "set-cookie": serialize("github_oauth_state", state, {
+        path: "/",
+        secure: process.env.NODE_ENV === "production",
+        httpOnly: true,
+        maxAge: 60 * 10,
+        sameSite: "lax",
+      }),
+    },
+  });
+}
+
+/**
+ * Github OAuth validate callback handler
+ *
+ * @link {@see https://lucia-auth.com/guides/oauth/basics#validate-callback}
+ */
+export async function luciaGithubCallbackHandler<Context extends Record<string | number | symbol, unknown>>(
+  request: Request,
+  _context?: Context,
+): Promise<Response> {
+  const cookies = parse(request.headers.get("cookie") ?? "");
+  const params = new URL(request.url).searchParams;
+  const code = params.get("code");
+  const state = params.get("state");
+  const storedState = cookies.github_oauth_state ?? null;
+
+  if (!code || !state || !storedState || state !== storedState) {
+    return new Response("Unauthorized Request", {
+      status: 401,
+    });
+  }
+
+  try {
+    const tokens = await github.validateAuthorizationCode(code);
+    const githubUserResponse = await fetch("https://api.github.com/user", {
+      headers: {
+        Authorization: `Bearer ${tokens.accessToken}`,
+      },
+    });
+    const githubUser = (await githubUserResponse.json()) as GitHubUser;
+
+    const existingAccount = getExistingAccount("github", githubUser.id) as DatabaseOAuthAccount | undefined;
+
+    if (existingAccount) {
+      const session = await lucia.createSession(
+        BATI.has("drizzle") ? existingAccount.userId : existingAccount.user_id,
+        {},
+      );
+      return new Response(null, {
+        status: 302,
+        headers: {
+          Location: "/",
+          "set-cookie": lucia.createSessionCookie(session.id).serialize(),
+        },
+      });
+    }
+
+    const userId = generateId(15);
+
+    if (BATI.has("drizzle")) {
+      drizzleDb.insert(userTable).values({ id: userId, username: githubUser.login }).run();
+      drizzleDb.insert(oauthAccountTable).values({ providerId: "github", providerUserId: githubUser.id, userId }).run();
```

Comment:
You should use [transaction](https://orm.drizzle.team/docs/transactions) for related queries.